### PR TITLE
UnixPB: Add support for Debian11/RISC-V

### PR DESCRIPTION
**Pull request by Willsparker**
Ref: #2123 

Adds support for debain11/RISC-V. Tested in QPC throughout the process, see the referenced issue for details.

##### Checklist

- [x] commit message has one of the [standard prefixes](https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [FAQ.md](https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [x] playbook changes run through [VPC](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptopenjdk.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access) [QPC](https://ci.adoptopenjdk.net/job/QEMUPlaybookCheck/267/ARCHITECTURE=riscv,OS=debian11,label=vagrant/console)
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

### CODE REVIEW

**Review by sxa — CHANGES_REQUESTED** [review]
Needs linter fixes, but good that we've got it though QPC now :-)

**Comment by Willsparker** [comment]
@sxa Yamlint has been fixed :-)

**Review by gdams — APPROVED** [review]

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml [hunk 1]
```diff
@@ -74,22 +74,31 @@
       when: ansible_distribution != "MacOSX"
     - role: adoptopenjdk_install
       jdk_version: 8
-      when: ansible_distribution != "Alpine"
+      when:
+        - ansible_distribution != "Alpine"
+        - ansible_architecture != "riscv64"
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml:79]: I can't help feel that these should be within the role, and the role should be kept up-to-date with which ones are available, but that's not really in scope for this PR

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Debian.yml [hunk 2]
```diff
@@ -33,10 +33,14 @@
...
     repo: "deb https://adoptopenjdk.jfrog.io/adoptopenjdk/deb {{ ansible_distribution_release }} main"
     update_cache: no
+  when:
+    - (ansible_architecture != "riscv64")
+  tags: patch_update
 
 - name: Add the ubuntu toolchain repository to apt
   apt_repository: repo='ppa:ubuntu-toolchain-r/test' update_cache=no
   when:
+    - (ansible_architecture != "armv7l") and (ansible_architecture != "riscv64")
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Debian.yml:43]: We seem to have `armv7l` here and on the next line
```suggestion
    - ansible_architecture != "riscv64"
```

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Debian.yml [hunk 3]
```diff
@@ -59,7 +59,7 @@ OpenJFX_Build_Tool_Packages:
   - libxxf86vm-dev
   - ruby
 
-gcc_7:                            # On all architectures but arm32
+gcc_7:                            # On all architectures but arm32 and riscv64
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Debian.yml:62]: Another situation where I feel we need a clean-up operation ... Ideally this should really depend on the distro and whether `gcc-7` is available, not fixed per architecture if possible, but not in scope for this PR.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Debian.yml [hunk 4]
```diff
@@ -90,9 +90,12 @@ Additional_Build_Tools_s390x:
 Additional_Build_Tools_aarch64:
   - libpng-dev
 
+Additional_Build_Tools_riscv64:
+  - locales
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Debian.yml:94]: Is this not done by https://github.com/adoptium/infrastructure/blob/793eac15bf0cb46ec92f67d83e6537d7ca9feac5/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Debian.yml#L196 ? Seems odd to have this as a RISC-V specific clause ... I feel the same is true of systemd-timesyncd

**Review by sxa — CHANGES_REQUESTED** [review]
A few more comments - only two I think need addressing/answered

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Debian.yml (hunk 4 above)
Comment (Willsparker) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Debian.yml:94]: `locales` is, from the task you linked ... Not sure about `systemd-timesyncd` ? The `NTP_Time` role doesn't install it

Comment (Willsparker) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Debian.yml]: I've removed `locales`, however I can't see anywhere else that `systemd-timesyncd` is installed. I took this particular piece of code from @gdams' #1483 , so `systemd-timesyncd` may have been required when setting up the `riscv` machines.

**Review by sxa — APPROVED** [review]

**Comment by karianna** [comment]
I assume we don't care about the failing Solaris GH Action?

**Comment by sxa** [comment]
> I assume we don't care about the failing Solaris GH Action?

We should care about it, but it's failing on every PR so it's not up to the owner of this PR specifically to fix it if it's not been introduced here.

**Pull request merged**